src/query/query_types.rs

```diff
@@ -1,12 +1,19 @@
 use std::cmp::Ordering;
 
 #[derive(Debug, PartialEq)]
-pub enum Lookup<'a> {
+pub enum PacketLookup<'a> {
     Name,
     Id,
     Parameter(&'a str),
 }
 
+#[derive(Debug, PartialEq)]
+pub enum Lookup<'a> {
+    Packet(PacketLookup<'a>),
+    This(&'a str),
+    Environment(&'a str),
+}
+
```

Comment:
Any thoughts on adding `this` and `environment` to the lookup instead of as a separate enum here. i.e. have

```
pub enum Lookup<'a> {
  Name,
  Id,
  Parameter(&'a str),
  This(&'a str),
  Environment(&'a str),
}
```

instead of as nested? 

I can see them logically being different things, were looking up from 3 different places. But wondering about that vs reducing some of the nesting. Just curious about why this way

Reply:
My first attempt looked like your suggestion, then I had to update the [`Packet::lookup_value`](https://github.com/mrc-ide/outpack_server/blob/6afeb75482474724f8180235fa763be4f703709c/src/query/query_eval.rs#L106) function which is scoped for a single packet, and therefore This and Environment make no sense as arguments to that function. Having the nested hierarchy makes that evident.

Reply:
ah yes that makes sense!